dashboards-observability/public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -539,6 +554,19 @@ export const Explorer = ({
         },
       })
     );
+
+    await dispatch(
+      changeQuery({
+        tabId,
+        query: {
+          [SELECTED_DATE_RANGE]: timeRange,
+        },
+      })
+    );
```

Comment:
query or data are the fields inside a dispatched event data, take look at here: https://github.com/opensearch-project/dashboards-observability/blob/2.x/dashboards-observability/public/components/event_analytics/redux/slices/query_slice.ts#L57 changeQuery looks for 'query', whereas changeDaterange looks for 'data', I don't remember why it's different but it can be any name you want, maybe ideally we should all use 'data'